thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/resources/PinotDataSourceResource.java

```diff
@@ -0,0 +1,84 @@
+package com.linkedin.thirdeye.datasource.pinot.resources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.common.base.Preconditions;
+import com.linkedin.thirdeye.datasource.ThirdEyeCacheRegistry;
+import com.linkedin.thirdeye.datasource.pinot.PinotQuery;
+import com.linkedin.thirdeye.datasource.pinot.PinotThirdEyeDataSource;
+import com.linkedin.thirdeye.datasource.pinot.resultset.ThirdEyeResultSet;
+import com.linkedin.thirdeye.datasource.pinot.resultset.ThirdEyeResultSetGroup;
+import com.linkedin.thirdeye.datasource.pinot.resultset.ThirdEyeResultSetSerializer;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.concurrent.ExecutionException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/pinot-data-source")
+@Produces(MediaType.APPLICATION_JSON)
+public class PinotDataSourceResource {
```

Comment:
this is very dangerous. please at least add documentation that this is for testing purposes only. It allows full access to pinot

Reply:
No, it won't. The accessibility of tables is set on Pinot. If a Pinot table is not accessable to TE, this endpoint won't make TE have superpower and access the table.